web-app/src/app/api/stripe/webhook/route.ts

```diff
@@ -0,0 +1,88 @@
+import { NextRequest, NextResponse } from "next/server";
+import Stripe from "stripe";
+
+import { getEnvSecrets } from "@/config/env.config";
+import { updateFiatTransactionStatus } from "@/lib/db/services/fiatTransaction.service";
+import { FiatTransactionStatus } from "@/prisma/generated/client";
+
+const stripe = new Stripe(getEnvSecrets().STRIPE_SECRET_KEY);
+
+const handleFiatTransactionFailed = async (
+  session: Stripe.Checkout.Session,
+) => {
+  console.log(`🔔  Payment failed for session ${session.id}`);
+
+  await updateFiatTransactionStatus(session.id, FiatTransactionStatus.FAILED);
+};
+
+const checkSessionPayment = async (session: Stripe.Checkout.Session) => {
+  console.log(`🔔  Check payment for session ${session.id}`);
+  const paymentStatus = session.payment_status;
+  if (paymentStatus !== "paid") {
+    console.error(
+      `🔔  Payment status is ${paymentStatus} for session ${session.id}`,
+    );
+    return;
+  }
+
+  await updateFiatTransactionStatus(
+    session.id,
+    FiatTransactionStatus.SUCCEEDED,
+  );
+};
+
+export async function POST(request: NextRequest) {
+  console.log("🔔  Stripe webhook received");
```

Comment:
I think we should remove unecessary logs?

Reply:
153fc7f066d92912551dfd7f9fa5aee4cadafafc